### ISSUE TITLE
Add overlay for left sidebar

### ISSUE THREAD

**Issue by arpansac** [commudle/commudle-ng#22]
We can only close the sidebar by clicking the menu button. But there should be an overlay on the mobile view.

**Comment by Parthchandratre** [comment]
Sir I am Interested in this issue.
But I am Beignner  so i will try to solve it

**Comment by arpansac** [comment]
Cool @Parthchandratre assigned to you

**Comment by Parthchandratre** [comment]
sir! I have made changes in HTML,scss File But I can't Find Where to make changes for Js  ( I Can't Find JS File)

**Comment by arpansac** [comment]
It's the .ts file for the component. I'm not sure if you'll need to make
changes to it!

Can you please share which files you've changed, and probably some
screenshots too

**Comment by Parthchandratre** [comment]
ok, May I sent Photos On Whats up

**Comment by arpansac** [comment]
Screenshots of the UI

**Comment by Parthchandratre** [comment]
The code added in html File
 `<div id="overlay">
           <div id="text">This Will Contain the text Which We Want to Show On Overlay`

Scss File=
`.overlay {
  position: fixed; 
  display: none; 
  width: 100%; 
  height: 100%; 
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5); 
  z-index: 2; 
  cursor: pointer;
}`  You Can Change This Code As Per You Sir!

I have Written It But Not Understanding How to Make it Functionable .

**Comment by arpansac** [comment]
Hey!
Have you tested it to work? Also, it should work on mobile

**Comment by Parthchandratre** [comment]
Sir! I Can't test on this project So, I tested it On My one of the project But it not Worked.
So my suggestion Is That 90% I think My code is right From Frontend So Can You Test it By your Side

**Comment by arpansac** [comment]
Are you facing a problem in setting up Commudle?

**Comment by arpansac** [comment]
Making this issue free to be occupied again.

**Comment by Sloth-Panda** [comment]
I would love to take this issue. Let me know.  There are also no DWoC labels on your issues can you please add the labels to them!

**Comment by arpansac** [comment]
Hi @Sloth-Panda, sure, I've assigned the issue to you. Please mention a timeline by when you will be able to complete it!
Let me know if you need my help in figuring it out.

**Comment by Sloth-Panda** [comment]
Just give me 2 days  or idk i will just try making it in sometimes today only may get completed very soon

**Comment by Sloth-Panda** [comment]
Hey, do I really need to download the angular CLI to work on this

**Comment by arpansac** [comment]
Yep! 
And take it easy, no pressures, let's decide a date of completion :)

> Hey, do I really need to download the angular CLI to work on this

**Comment by Sloth-Panda** [comment]
Why can't I simply modify the scss file using vscode ??

**Comment by arpansac** [comment]
As long as you can test it and send over the screenshots, it works

> Why can't I simply modify the scss file using vscode ??

**Comment by Sloth-Panda** [comment]
There won't be any issue with merging pr if it works fine right? If I do it in the above-mentioned way?

**Comment by arpansac** [comment]
> There won't be any issue with merging pr if it works fine right? If I do it in the above-mentioned way?

As long as it works fine and the code's clean, it's cool. We will need the screenshots.

**Comment by Sloth-Panda** [comment]
![Screenshot (7)](https://user-images.githubusercontent.com/70213384/103239467-92545880-4973-11eb-91d7-75099854a325.png)
Whenever I try to install angular/cli using   git install -g @angular/cli  It shows some problem like this
Please suggest what to do

**Comment by Ashwin-op** [comment]
> ![Screenshot (7)](https://user-images.githubusercontent.com/70213384/103239467-92545880-4973-11eb-91d7-75099854a325.png)
> Whenever I try to install angular/cli using git install -g @angular/cli It shows some problem like this
> Please suggest what to do

Angular CLI can't be called with `angular` or `angular/cli`. Try `ng v` to see the version and check if angular cli is installed in your system

**Comment by Sloth-Panda** [comment]
![Screenshot (9)](https://user-images.githubusercontent.com/70213384/103259881-46300500-49c1-11eb-8c41-d36550a943aa.png)

On my system angular cli is installed, after that, I ran npm install, and after that ng serve but when i open the go over the http://localhost:4200/  it just says page mot found why's that. i have good internet connection though

**Comment by arpansac** [comment]
> ![Screenshot (9)](https://user-images.githubusercontent.com/70213384/103259881-46300500-49c1-11eb-8c41-d36550a943aa.png)
> 
> On my system angular cli is installed, after that, I ran npm install, and after that ng serve but when i open the go over the http://localhost:4200/ it just says page mot found why's that. i have good internet connection though

Great, you're on the right track. Go to environment.ts inside commudle-admin/src and check if it looks like this:

`export const environment = {
  production: false,
  // app_url: 'http://localhost:4200',
  // base_url: 'http://localhost:3000',
  // action_cable_url: 'http://localhost:3000/live',
  // auth_cookie_name: 'dev_commudle_user_auth',
  // zoom_call_server_url: 'http://localhost:4000',

  session_cookie_name: 'cmdle_sess',

  app_url: 'https://commudle.com',
  base_url: 'https://json.commudle.com',
  action_cable_url: 'https://json.commudle.com/live',
  auth_cookie_name: 'commudle_user_auth',
  zoom_call_server_url: 'https://zoom.commudle.com',
};`

If not then make it this way.
If it looks like this then if you can share a screenshot here...

**Comment by Sloth-Panda** [comment]
![Screenshot (10)](https://user-images.githubusercontent.com/70213384/103272120-075e7700-49e2-11eb-8cfa-7b0f868fc1e4.png)

is this the place if not can you tell me the exact path where to put this angualr packages's path

**Comment by arpansac** [comment]
> ![Screenshot (10)](https://user-images.githubusercontent.com/70213384/103272120-075e7700-49e2-11eb-8cfa-7b0f868fc1e4.png)
> 
> is this the place if not can you tell me the exact path where to put this angualr packages's path

Sorry, I'm a little confused....
There's no need to change any package paths...

**Comment by Sloth-Panda** [comment]
`export const environment = {
production: false,
// app_url: 'http://localhost:4200',
// base_url: 'http://localhost:3000',
// action_cable_url: 'http://localhost:3000/live',
// auth_cookie_name: 'dev_commudle_user_auth',
// zoom_call_server_url: 'http://localhost:4000',

session_cookie_name: 'cmdle_sess',

app_url: 'https://commudle.com',
base_url: 'https://json.commudle.com',
action_cable_url: 'https://json.commudle.com/live',
auth_cookie_name: 'commudle_user_auth',
zoom_call_server_url: 'https://zoom.commudle.com',
};`



I meant where to put this it's path

**Comment by arpansac** [comment]
> `export const environment = {
> production: false,
> // app_url: 'http://localhost:4200',
> // base_url: 'http://localhost:3000',
> // action_cable_url: 'http://localhost:3000/live',
> // auth_cookie_name: 'dev_commudle_user_auth',
> // zoom_call_server_url: 'http://localhost:4000',
> 
> session_cookie_name: 'cmdle_sess',
> 
> app_url: 'https://commudle.com',
> base_url: 'https://json.commudle.com',
> action_cable_url: 'https://json.commudle.com/live',
> auth_cookie_name: 'commudle_user_auth',
> zoom_call_server_url: 'https://zoom.commudle.com',
> };`
> 
> I meant where to put this it's path

Oh that, sure, it's this file https://github.com/commudle/commudle-ng/blob/development/projects/commudle-admin/src/environments/environment.ts

Also, I'd suggest exploring the folder structure a little more on how things are aligned, it'll help you navigate faster, else, I'm here to help as well.

**Comment by Sloth-Panda** [comment]
![Screenshot (11)](https://user-images.githubusercontent.com/70213384/103276750-6b3a6d00-49ed-11eb-9629-fa7cac9915e5.png)

It's showing like this as you told but still when i tried to run this project on localhost same issue
Do I need to start XAMPP inorder to make this project run on localhost

**Comment by arpansac** [comment]
Nope, it should directly run after you do npm install and then ng s

**Comment by arpansac** [comment]
Hey @Sloth-Panda, Happy New Year!
Any progress with this?

**Comment by Sloth-Panda** [comment]
I am having aproblem with the setup let me try again and will let you know.

**Comment by arpansac** [comment]
Cool, otherwise let me know we can get on a screenshare and I'll help you
fix it.

**Comment by Sloth-Panda** [comment]
Yes, sir the localhost at 4200 now leads to ur site having some issue previously so i will look into the issues now will reply you back soon

I tried logging in now but I can't find any cookie namely  commudle_user_auth
![Screenshot (20)](https://user-images.githubusercontent.com/70213384/103450899-f912a280-4ce2-11eb-8c6f-53f7179f0c74.png)


Or else it asking to remove the adBlocker or antivirus extensions. so how can i signin

![Screenshot (21)](https://user-images.githubusercontent.com/70213384/103450939-9b328a80-4ce3-11eb-9114-8cf91c03e7ae.png)
![Screenshot (22)](https://user-images.githubusercontent.com/70213384/103450940-9ff73e80-4ce3-11eb-8ff5-a42b03781931.png)


I found no such cookie

**Comment by Sloth-Panda** [comment]
sir can you tell me how can I login

**Comment by arpansac** [comment]
> sir can you tell me how can I login

Sure, let me put down the steps:
- Click the sign in button on auther.commudle.com
- It'll take you to google for signing in
- Then it will redirect you back to commudle.com
- Now you will find the cookie, just copy and paste into the tab where you are running localhost :)

**Comment by arpansac** [comment]
@Sloth-Panda please update on the progress.

**Comment by sakshikhachane** [comment]
@arpansac I would like to work on this issue. Is it available?

**Comment by arpansac** [comment]
@Sloth-Panda please update in case you are working on this

**Comment by Sloth-Panda** [comment]
She can go ahead

**Comment by arpansac** [comment]
> She can go ahead

Thanks for the clarity, wish you had updated earlier if you weren't interested 

@sakshikhachane please setup the code on your system. Once done I'll assign this to you.

**Comment by sakshikhachane** [comment]
@arpansac I am done with the setup and it's running on my system.

**Comment by arpansac** [comment]
Implemented more than a year ago. Closing this now.
